<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Computationally identify biologically relevant alterations — run_CIBRA • CIBRA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Computationally identify biologically relevant alterations — run_CIBRA"><meta name="description" content="run_CIBRA calculates an impact score from provided expression data and
defined definitions"><meta property="og:description" content="run_CIBRA calculates an impact score from provided expression data and
defined definitions"><meta property="og:image" content="https://ait4life-uu.github.io/CIBRA/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CIBRA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/alteration_type_refinement.html">Refining alterations on mutation type</a></li>
    <li><a class="dropdown-item" href="../articles/genome_wide_screen.html">Genome wide screen using CIBRA</a></li>
    <li><a class="dropdown-item" href="../articles/genomic_location_refinement.html">Refining alterations on genomic location</a></li>
    <li><a class="dropdown-item" href="../articles/impact_assessment_genomic_alteration.html">CIBRA impact score for a genomic alteration</a></li>
    <li><a class="dropdown-item" href="../articles/reference_permutation_distribution.html">reference permutation distribution</a></li>
    <li><a class="dropdown-item" href="../articles/similarity_assessment.html">CIBRA similarity score</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Computationally identify biologically relevant alterations</h1>

      <div class="d-none name"><code>run_CIBRA.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>run_CIBRA</code> calculates an impact score from provided expression data and
defined definitions</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">run_CIBRA</span><span class="op">(</span></span>
<span>  <span class="va">data</span>,</span>
<span>  <span class="va">definition_matrix</span>,</span>
<span>  <span class="va">columns</span>,</span>
<span>  <span class="va">control_definition</span>,</span>
<span>  confidence <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  iterations <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"DESeq2"</span>,</span>
<span>  parallel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  speedup <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  min_n <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  outputDir <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  sample_column <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-data">data<a class="anchor" aria-label="anchor" href="#arg-data"></a></dt>
<dd><p>RNA count dataframe with genes as rows and samples as columns</p></dd>


<dt id="arg-definition-matrix">definition_matrix<a class="anchor" aria-label="anchor" href="#arg-definition-matrix"></a></dt>
<dd><p>design dataframe of the comparisons to make, binary
columns (factors) with an indication of the groups to test and samples as rownames.</p></dd>


<dt id="arg-columns">columns<a class="anchor" aria-label="anchor" href="#arg-columns"></a></dt>
<dd><p>columns from the definition_matrix to assess with respect to
the control definition</p></dd>


<dt id="arg-control-definition">control_definition<a class="anchor" aria-label="anchor" href="#arg-control-definition"></a></dt>
<dd><p>Definition term that will be used as reference for
the comparison (e.g. WT)</p></dd>


<dt id="arg-confidence">confidence<a class="anchor" aria-label="anchor" href="#arg-confidence"></a></dt>
<dd><p>alpha threshold to calculate the proportion (default is 0.1)</p></dd>


<dt id="arg-iterations">iterations<a class="anchor" aria-label="anchor" href="#arg-iterations"></a></dt>
<dd><p>number of iterations for random permutation (recommendation is 50 iterations per definition)</p></dd>


<dt id="arg-covariates">covariates<a class="anchor" aria-label="anchor" href="#arg-covariates"></a></dt>
<dd><p>list of column names from the definition matrix to use as covariates (supported only with DESeq2)</p></dd>


<dt id="arg-method">method<a class="anchor" aria-label="anchor" href="#arg-method"></a></dt>
<dd><p>method to use for differential expression analysis.
options: 'DESeq2', 'edgeR' and 'limma'.
Defaults to 'DESeq2'</p></dd>


<dt id="arg-parallel">parallel<a class="anchor" aria-label="anchor" href="#arg-parallel"></a></dt>
<dd><p>boolean state to operate in parallel mode. Defaults to
<code>FALSE</code></p></dd>


<dt id="arg-speedup">speedup<a class="anchor" aria-label="anchor" href="#arg-speedup"></a></dt>
<dd><p>boolean state to run DESeq2 in speedup mode. Defaults to
<code>FALSE</code></p></dd>


<dt id="arg-min-n">min_n<a class="anchor" aria-label="anchor" href="#arg-min-n"></a></dt>
<dd><p>minimal number of samples to consider for analysis. Defaults to
the recommended minimum number of samples: <code>10</code></p></dd>


<dt id="arg-outputdir">outputDir<a class="anchor" aria-label="anchor" href="#arg-outputdir"></a></dt>
<dd><p>path for result files, if <code>FALSE</code>, no results will be
written. Defaults to <code>FALSE</code></p></dd>


<dt id="arg-sample-column">sample_column<a class="anchor" aria-label="anchor" href="#arg-sample-column"></a></dt>
<dd><p>name of the column containing the sample names, if NULL, rownames will be used</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>returns a list containing a dataframe with the signal measures and
associated metadata and associated results generated by differential
expression analysis</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: dplyr</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘dplyr’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:stats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     filter, lag</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     intersect, setdiff, setequal, union</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># load transcription data and genomic alterations</span></span></span>
<span class="r-in"><span><span class="va">count_data</span> <span class="op">&lt;-</span> <span class="fu">CIBRA</span><span class="fu">::</span><span class="va">TCGA_CRC_rna_data</span></span></span>
<span class="r-in"><span><span class="va">mutation_profile</span> <span class="op">&lt;-</span> <span class="fu">CIBRA</span><span class="fu">::</span><span class="va">TCGA_CRC_non_silent_mutation_profile</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># select BRAF as example</span></span></span>
<span class="r-in"><span><span class="va">goi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"BRAF"</span>, <span class="st">"APC"</span>, <span class="st">"TP53"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># filter mutation profile for only selected genes</span></span></span>
<span class="r-in"><span><span class="va">definition_matrix</span> <span class="op">&lt;-</span> <span class="va">mutation_profile</span><span class="op">[</span><span class="va">goi</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># select only 30 samples for the cases (BRAF mutant) and controls (BRAF WT) for the sake of speed of the example</span></span></span>
<span class="r-in"><span><span class="co"># you can remove this limit if you want to run the full comparison</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">definition_matrix</span> <span class="op">&lt;-</span> <span class="va">definition_matrix</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">'sample'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">BRAF</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/sample_n.html" class="external-link">sample_n</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">'sample'</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#' # fix sample names for intersect between the two datatypes</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">definition_matrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html" class="external-link">str_replace_all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">definition_matrix</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="st">"."</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># select intersect between definition matrix and count data</span></span></span>
<span class="r-in"><span><span class="va">sample_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">definition_matrix</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">count_data</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># select data and definition matrix as the intersect</span></span></span>
<span class="r-in"><span><span class="va">count_data</span> <span class="op">&lt;-</span> <span class="va">count_data</span><span class="op">[</span><span class="va">sample_list</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">definition_matrix</span> <span class="op">&lt;-</span> <span class="va">definition_matrix</span><span class="op">[</span><span class="va">sample_list</span>,<span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># select BRAF</span></span></span>
<span class="r-in"><span><span class="va">columns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"BRAF"</span><span class="op">)</span> <span class="co"># if more column were provided in the definition_matrix</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># set up parameters for CIBRA</span></span></span>
<span class="r-in"><span><span class="va">control_definition</span> <span class="op">&lt;-</span> <span class="st">"WT"</span> <span class="co"># reference condition</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">confidence</span> <span class="op">&lt;-</span> <span class="fl">0.1</span> <span class="co"># confidence threshold for the proportion calculation</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># set permutation to 0 to skip permutations</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">iterations</span> <span class="op">=</span> <span class="fl">0</span> <span class="co"># number of permutation iterations to be performed per condition</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># register(SnowParam(2)) # set number of cores to be used for parallel processing</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># run CIBRA with default DE analysis method (DESeq2)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">CIBRA_res</span> <span class="op">&lt;-</span> <span class="fu">run_CIBRA</span><span class="op">(</span><span class="va">count_data</span>, <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">definition_matrix</span><span class="op">)</span>, columns <span class="op">=</span> <span class="va">columns</span>, control_definition <span class="op">=</span> <span class="va">control_definition</span>, confidence <span class="op">=</span> <span class="va">confidence</span>, iterations <span class="op">=</span> <span class="va">iterations</span>, parallel <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="co"># set parallel to TRUE for parallel processing</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "SNV"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "BRAF" "SNV" </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating size factors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating dispersions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> gene-wise dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> mean-dispersion relationship</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> final dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fitting model and testing</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -- replacing outliers and refitting for 2735 genes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -- DESeq argument 'minReplicatesForReplace' = 7 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -- original counts are preserved in counts(dds)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating dispersions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fitting model and testing</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># calculate the pvalue by comparing to a reference distribution generated with DESeq2</span></span></span>
<span class="r-in"><span><span class="va">perm_data</span> <span class="op">&lt;-</span> <span class="fu">CIBRA</span><span class="fu">::</span><span class="va">perm_dist_crc_tcga</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># use the gamma test to also calculate a fitted p-value</span></span></span>
<span class="r-in"><span><span class="va">CIBRA_res_stat</span> <span class="op">&lt;-</span>  <span class="fu"><a href="permutation_test.html">permutation_test</a></span><span class="op">(</span><span class="va">CIBRA_res</span><span class="op">$</span><span class="va">CIBRA_results</span>, <span class="va">perm_data</span>, test <span class="op">=</span> <span class="st">"gamma"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Soufyan Lakbir.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

