<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Refining alterations on genomic location • CIBRA</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Refining alterations on genomic location">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CIBRA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/alteration_type_refinement.html">Refining alterations on mutation type</a></li>
    <li><a class="dropdown-item" href="../articles/genome_wide_screen.html">Genome wide screen using CIBRA</a></li>
    <li><a class="dropdown-item" href="../articles/genomic_location_refinement.html">Refining alterations on genomic location</a></li>
    <li><a class="dropdown-item" href="../articles/impact_assessment_genomic_alteration.html">CIBRA impact score for a genomic alteration</a></li>
    <li><a class="dropdown-item" href="../articles/reference_permutation_distribution.html">reference permutation distribution</a></li>
    <li><a class="dropdown-item" href="../articles/similarity_assessment.html">CIBRA similarity score</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Refining alterations on genomic location</h1>
            
      

      <div class="d-none name"><code>genomic_location_refinement.Rmd</code></div>
    </div>

    
    
<p>Similarly to the vignette titled: “Refining alterations on mutation
type”, in this vignette we will refine on the genomic location of the
alteration.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ait4life-uu.github.io/CIBRA/">CIBRA</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Bioconductor/BiocParallel" class="external-link">BiocParallel</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></code></pre></div>
<p>We will explore the impact of variants affecting different exons in
APC. To scan through genomic locations using CIBRA, the definition
matrix can be set up in an extended format where samples can be assigned
to multiple definitions. In this setting, it is important to have a
sample column.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load data</span></span>
<span><span class="va">count_data</span> <span class="op">&lt;-</span> <span class="fu">CIBRA</span><span class="fu">::</span><span class="va">TCGA_CRC_rna_data</span> <span class="co"># rna count data</span></span>
<span></span>
<span><span class="co"># genomic alteration profile</span></span>
<span><span class="va">definition_matrix</span> <span class="op">&lt;-</span> <span class="fu">CIBRA</span><span class="fu">::</span><span class="va">TCGA_APC_exon_definition_transcript_201</span></span>
<span></span>
<span><span class="co"># remove rownames column</span></span>
<span><span class="va">definition_matrix</span><span class="op">$</span><span class="va">...1</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span></span>
<span><span class="co"># remove letter to match sample names of genomic metrics</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">count_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/make.names.html" class="external-link">make.names</a></span><span class="op">(</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_sub.html" class="external-link">str_sub</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">count_data</span><span class="op">)</span>, end <span class="op">=</span> <span class="op">-</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create barcode from sample ids</span></span>
<span><span class="va">definition_matrix</span><span class="op">$</span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="va">definition_matrix</span><span class="op">$</span><span class="va">samples</span>, <span class="fl">1</span>, <span class="fl">15</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create valid names to match count data</span></span>
<span><span class="va">definition_matrix</span> <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html" class="external-link">mutate_all</a></span><span class="op">(</span><span class="va">definition_matrix</span>, <span class="va">make.names</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># correct the colnames of the definition matrix (CIBRA takes only proper colnames)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">definition_matrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/make.names.html" class="external-link">make.names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">definition_matrix</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We will select the column with the exon definitions and exclude the
sample column. In this way, we will only test the column we have
selected with the definitions. Because we are running the definition
matrix in the extended mode, we need to indicate the sample column name.
In this way CIBRA is aware that there can be duplicate sample names. It
will run CIBRA for each definition compared to the control
definition.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># select columns with definition</span></span>
<span><span class="va">columns</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">definition_matrix</span><span class="op">)</span></span>
<span><span class="va">columns</span> <span class="op">=</span> <span class="va">columns</span><span class="op">[</span><span class="va">columns</span> <span class="op">!=</span> <span class="st">"samples"</span><span class="op">]</span> <span class="co"># exclude sample column</span></span>
<span></span>
<span><span class="co"># set up parameters for CIBRA</span></span>
<span><span class="va">control_definition</span> <span class="op">&lt;-</span> <span class="st">"no.SNV"</span> <span class="co"># reference condition</span></span>
<span><span class="va">confidence</span> <span class="op">&lt;-</span> <span class="fl">0.1</span> <span class="co"># confidence threshold for the proportion calculation</span></span>
<span><span class="co"># set permutation to 0 to skip permutations</span></span>
<span><span class="va">iterations</span> <span class="op">=</span> <span class="fl">0</span> <span class="co"># number of permutation iterations to be performed per condition</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">register</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/SnowParam-class.html" class="external-link">SnowParam</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span> <span class="co"># set number of cores to be used</span></span>
<span>  </span>
<span><span class="co"># select column with definitions</span></span>
<span><span class="va">group</span> <span class="op">=</span> <span class="va">definition_matrix</span></span>
<span></span>
<span><span class="co"># run CIBRA</span></span>
<span><span class="va">CIBRA_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_CIBRA.html">run_CIBRA</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">count_data</span>,definition_matrix <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">definition_matrix</span><span class="op">)</span>,columns <span class="op">=</span> <span class="va">columns</span>,</span>
<span>                       control_definition <span class="op">=</span> <span class="va">control_definition</span>, confidence <span class="op">=</span> <span class="va">confidence</span>, </span>
<span>                       iterations <span class="op">=</span> <span class="va">iterations</span>,</span>
<span>                       sample_column <span class="op">=</span> <span class="st">"samples"</span>,</span>
<span>                       parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "X16"        "X12"        "X2"         "X9"         "X7"        </span></span>
<span><span class="co">#&gt;  [6] "X10"        "X13"        "X14"        "X8"         "X3"        </span></span>
<span><span class="co">#&gt; [11] "X5"         "X15"        "X6"         "X4"         "non_coding"</span></span>
<span><span class="co">#&gt; [1] "Transcript.APC.201" "X16"</span></span>
<span><span class="co">#&gt; [1] "Transcript.APC.201" "X12"               </span></span>
<span><span class="co">#&gt; [1] "number of cases &lt; 10"</span></span>
<span><span class="co">#&gt; [1] "Transcript.APC.201" "X2"                </span></span>
<span><span class="co">#&gt; [1] "number of cases &lt; 10"</span></span>
<span><span class="co">#&gt; [1] "Transcript.APC.201" "X9"</span></span>
<span><span class="co">#&gt; [1] "Transcript.APC.201" "X7"</span></span>
<span><span class="co">#&gt; [1] "Transcript.APC.201" "X10"</span></span>
<span><span class="co">#&gt; [1] "Transcript.APC.201" "X13"               </span></span>
<span><span class="co">#&gt; [1] "number of cases &lt; 10"</span></span>
<span><span class="co">#&gt; [1] "Transcript.APC.201" "X14"</span></span>
<span><span class="co">#&gt; [1] "Transcript.APC.201" "X8"                </span></span>
<span><span class="co">#&gt; [1] "number of cases &lt; 10"</span></span>
<span><span class="co">#&gt; [1] "Transcript.APC.201" "X3"                </span></span>
<span><span class="co">#&gt; [1] "number of cases &lt; 10"</span></span>
<span><span class="co">#&gt; [1] "Transcript.APC.201" "X5"                </span></span>
<span><span class="co">#&gt; [1] "number of cases &lt; 10"</span></span>
<span><span class="co">#&gt; [1] "Transcript.APC.201" "X15"</span></span>
<span><span class="co">#&gt; [1] "Transcript.APC.201" "X6"</span></span>
<span><span class="co">#&gt; [1] "Transcript.APC.201" "X4"                </span></span>
<span><span class="co">#&gt; [1] "number of cases &lt; 10"</span></span>
<span><span class="co">#&gt; [1] "Transcript.APC.201" "non_coding"        </span></span>
<span><span class="co">#&gt; [1] "number of cases &lt; 10"</span></span></code></pre></div>
<p>The results are now structured as the exon number (def) and the
column we selected as “col”. The number of cases is provided by the
investigated column.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># print the report table</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">CIBRA_res</span><span class="op">$</span><span class="va">CIBRA_results</span>, format <span class="op">=</span> <span class="st">"html"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr>
<th style="text-align:left;">
def
</th>
<th style="text-align:left;">
col
</th>
<th style="text-align:left;">
proportion
</th>
<th style="text-align:left;">
sign_area
</th>
<th style="text-align:left;">
investigated
</th>
<th style="text-align:left;">
control
</th>
<th style="text-align:left;">
proportion_rnd_avg
</th>
<th style="text-align:left;">
proportion_rnd_sd
</th>
<th style="text-align:left;">
sign_area_rnd_avg
</th>
<th style="text-align:left;">
sign_area_rnd_sd
</th>
<th style="text-align:left;">
iterations
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
X16
</td>
<td style="text-align:left;">
Transcript.APC.201
</td>
<td style="text-align:left;">
0.4096
</td>
<td style="text-align:left;">
0.4179
</td>
<td style="text-align:left;">
301
</td>
<td style="text-align:left;">
113
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
X9
</td>
<td style="text-align:left;">
Transcript.APC.201
</td>
<td style="text-align:left;">
0.1689
</td>
<td style="text-align:left;">
0.1184
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
113
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
X7
</td>
<td style="text-align:left;">
Transcript.APC.201
</td>
<td style="text-align:left;">
0.2628
</td>
<td style="text-align:left;">
0.2633
</td>
<td style="text-align:left;">
24
</td>
<td style="text-align:left;">
113
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
X10
</td>
<td style="text-align:left;">
Transcript.APC.201
</td>
<td style="text-align:left;">
0.1461
</td>
<td style="text-align:left;">
0.0728
</td>
<td style="text-align:left;">
13
</td>
<td style="text-align:left;">
113
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
X14
</td>
<td style="text-align:left;">
Transcript.APC.201
</td>
<td style="text-align:left;">
0.2445
</td>
<td style="text-align:left;">
0.2406
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
113
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
X15
</td>
<td style="text-align:left;">
Transcript.APC.201
</td>
<td style="text-align:left;">
0.2188
</td>
<td style="text-align:left;">
0.2144
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
113
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
X6
</td>
<td style="text-align:left;">
Transcript.APC.201
</td>
<td style="text-align:left;">
0.2405
</td>
<td style="text-align:left;">
0.2338
</td>
<td style="text-align:left;">
20
</td>
<td style="text-align:left;">
113
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
0
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate the pvalue by comparing to a reference distribution</span></span>
<span><span class="va">perm_data</span> <span class="op">&lt;-</span> <span class="fu">CIBRA</span><span class="fu">::</span><span class="va">perm_dist_crc_tcga</span></span>
<span></span>
<span><span class="co"># use the gamma test to also calculate a fitted p-value</span></span>
<span><span class="va">CIBRA_res_stat</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/permutation_test.html">permutation_test</a></span><span class="op">(</span><span class="va">CIBRA_res</span><span class="op">$</span><span class="va">CIBRA_results</span>, <span class="va">perm_data</span>, test <span class="op">=</span> <span class="st">"gamma"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># print the report table</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">CIBRA_res_stat</span><span class="op">$</span><span class="va">results</span>, format <span class="op">=</span> <span class="st">"html"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr>
<th style="text-align:left;">
def
</th>
<th style="text-align:left;">
col
</th>
<th style="text-align:left;">
proportion
</th>
<th style="text-align:left;">
sign_area
</th>
<th style="text-align:left;">
investigated
</th>
<th style="text-align:left;">
control
</th>
<th style="text-align:left;">
proportion_rnd_avg
</th>
<th style="text-align:left;">
proportion_rnd_sd
</th>
<th style="text-align:left;">
sign_area_rnd_avg
</th>
<th style="text-align:left;">
sign_area_rnd_sd
</th>
<th style="text-align:left;">
iterations
</th>
<th style="text-align:right;">
perm_test_prop
</th>
<th style="text-align:right;">
perm_test_area
</th>
<th style="text-align:right;">
pvalue_prop
</th>
<th style="text-align:right;">
pvalue_sign_area
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
X16
</td>
<td style="text-align:left;">
Transcript.APC.201
</td>
<td style="text-align:left;">
0.4096
</td>
<td style="text-align:left;">
0.4179
</td>
<td style="text-align:left;">
301
</td>
<td style="text-align:left;">
113
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000248
</td>
<td style="text-align:right;">
0.0010112
</td>
</tr>
<tr>
<td style="text-align:left;">
X9
</td>
<td style="text-align:left;">
Transcript.APC.201
</td>
<td style="text-align:left;">
0.1689
</td>
<td style="text-align:left;">
0.1184
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
113
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
0.3412698
</td>
<td style="text-align:right;">
0.2839506
</td>
<td style="text-align:right;">
0.3767298
</td>
<td style="text-align:right;">
0.3392811
</td>
</tr>
<tr>
<td style="text-align:left;">
X7
</td>
<td style="text-align:left;">
Transcript.APC.201
</td>
<td style="text-align:left;">
0.2628
</td>
<td style="text-align:left;">
0.2633
</td>
<td style="text-align:left;">
24
</td>
<td style="text-align:left;">
113
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
0.0370370
</td>
<td style="text-align:right;">
0.0352734
</td>
<td style="text-align:right;">
0.0210423
</td>
<td style="text-align:right;">
0.0251315
</td>
</tr>
<tr>
<td style="text-align:left;">
X10
</td>
<td style="text-align:left;">
Transcript.APC.201
</td>
<td style="text-align:left;">
0.1461
</td>
<td style="text-align:left;">
0.0728
</td>
<td style="text-align:left;">
13
</td>
<td style="text-align:left;">
113
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
0.5476190
</td>
<td style="text-align:right;">
0.5291005
</td>
<td style="text-align:right;">
0.5774369
</td>
<td style="text-align:right;">
0.6315102
</td>
</tr>
<tr>
<td style="text-align:left;">
X14
</td>
<td style="text-align:left;">
Transcript.APC.201
</td>
<td style="text-align:left;">
0.2445
</td>
<td style="text-align:left;">
0.2406
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
113
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
0.0537919
</td>
<td style="text-align:right;">
0.0449735
</td>
<td style="text-align:right;">
0.0415007
</td>
<td style="text-align:right;">
0.0391983
</td>
</tr>
<tr>
<td style="text-align:left;">
X15
</td>
<td style="text-align:left;">
Transcript.APC.201
</td>
<td style="text-align:left;">
0.2188
</td>
<td style="text-align:left;">
0.2144
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
113
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
0.1111111
</td>
<td style="text-align:right;">
0.0643739
</td>
<td style="text-align:right;">
0.0988729
</td>
<td style="text-align:right;">
0.0646409
</td>
</tr>
<tr>
<td style="text-align:left;">
X6
</td>
<td style="text-align:left;">
Transcript.APC.201
</td>
<td style="text-align:left;">
0.2405
</td>
<td style="text-align:left;">
0.2338
</td>
<td style="text-align:left;">
20
</td>
<td style="text-align:left;">
113
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
0.0599647
</td>
<td style="text-align:right;">
0.0493827
</td>
<td style="text-align:right;">
0.0478327
</td>
<td style="text-align:right;">
0.0446955
</td>
</tr>
</tbody>
</table></main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Soufyan Lakbir.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
